### ISSUE TITLE
Migrate code to .net core

### ISSUE THREAD

**Issue by mgasparel** [yarus/sc2buildmaker-win#1]
I was going to take a stab at getting this app updated to .net core 3.1

What do you think about splitting this into two phases

1. Update class libraries to netstandard2 and update WinForms application to NET 4.7.2
2. Convert WinForms application to Xamarin.Forms

Phase 1 would be pretty straightforward and could be done in an evening. Phase 2 would take considerably more work.

**Comment by yarus** [comment]
Yeah, that could be the way to go. Though, we definitely need to understand what kind of goal we set for this migration. WinForms app's purpose is only to test new build configurations. I also used it to generate new Version file after making adjustments but this case could be easily moved to separate .NET Core console app. So I think, I would suggest to adjust phase 1 and add development of new .NET Core console app which would generate Version file based on current configuration.

**Comment by mgasparel** [comment]
> development of new .NET Core console app which would generate Version file based on current configuration.

I don't really understand how Version files are generated. I took a look through the UI code and I see the "Build Maker" application, but I don't see any views related to updating the version settings.

**Comment by yarus** [comment]
> > development of new .NET Core console app which would generate Version file based on current configuration.
> 
> I don't really understand how Version files are generated. I took a look through the UI code and I see the "Build Maker" application, but I don't see any views related to updating the version settings.

I use UnitTests project to do necessary changes:
![image](https://user-images.githubusercontent.com/1632102/84053252-51d2c480-a9ba-11ea-8549-de090c47b120.png)

So when I need to adjust some build item description, I go to UnitTests project, LOTV folder, search for specific item like SVC and do adjustments in static description. The next stage is to generate new Version file based on that change. In order to do that I use SC2.Win.UI application. Take a look sc2buildmaker-win\SC2.Win.UI\BuildMakerApplicationContext.cs - line 17:
mUiWorkflow = CreateTestUiWorkflow();

This CreateTestUiWorkflow method call will load everything from UnitTests project and then serialize it into Version file.

**Comment by yarus** [comment]
It would be much better if we rework SC2.Win.UI in order to add visual Version file configuration editor. It could be very simple actually - all static fields could be edited via simple input controls. It could be more difficult for order/produce requirements and actions though.

**Comment by mgasparel** [comment]
Thanks for the guidance. I took another look and it seems relatively straightforward. I will start looking at building a UI for this.
